ios/brave-ios/Sources/AIChat/ModelView/AIChatModelView.swift

```diff
@@ -0,0 +1,303 @@
+import Shared
+
+public class AIChatViewModel: NSObject, AIChatDelegate, ObservableObject {
+  private var api: AIChat!
+  private let webView: WKWebView?
+  private let script: any AIChatJavascript.Type
+  public var querySubmited: String?
+  
+  @Published var siteInfo: AiChat.SiteInfo?
+  @Published var premiumStatus: AiChat.PremiumStatus = .inactive
+  @Published var suggestedQuestions: [String] = []
+  @Published var conversationHistory: [AiChat.ConversationTurn] = []
+  @Published var models: [AiChat.Model] = []
+  @Published var currentModel: AiChat.Model!
+  
+  @Published var requestInProgress: Bool = false
+  @Published var apiError: AiChat.APIError = .none
+   
+  public var isContentAssociationPossible: Bool {
+    return webView?.url?.isWebPage(includeDataURIs: true) == true
+  }
+  
+  public var shouldSendPageContents: Bool {
+    get {
+      return api.shouldSendPageContents
+    }
+    
+    set {
+      objectWillChange.send()
+      api.shouldSendPageContents = newValue
+    }
+  }
+  
+  public var shouldShowPremiumPrompt: Bool {
+    get {
+      return api.canShowPremiumPrompt
+    }
+    
+    set {  // swiftlint:disable:this unused_setter_value
+      objectWillChange.send()
+      if !newValue {
+        api.dismissPremiumPrompt()
+      }
+    }
+  }
+  
+  public var shouldShowTermsAndConditions: Bool {
+    !api.conversationHistory.isEmpty && !api.isAgreementAccepted
+  }
+  
+  public var shouldShowSuggestions: Bool {
+    api.currentAPIError == .none &&
+    api.isAgreementAccepted &&
+    api.shouldSendPageContents && (
+      !api.suggestedQuestions.isEmpty ||
+      api.suggestionsStatus == .canGenerate ||
+      api.suggestionsStatus == .isGenerating
+    )
+  }
+  
+  public var shouldShowGenerateSuggestionsButton: Bool {
+    api.suggestionsStatus == .canGenerate || api.suggestionsStatus == .isGenerating
+  }
+  
+  public var suggestionsStatus: AiChat.SuggestionGenerationStatus {
+    api.suggestionsStatus
+  }
+  
+  public var isAgreementAccepted: Bool {
+    get {
+      return api.isAgreementAccepted
+    }
+    
+    set {
+      objectWillChange.send()
+      api.isAgreementAccepted = newValue
+      api.setConversationActive(newValue)
+    }
+  }
+  
+  public init(braveCore: BraveCoreMain,
+              webView: WKWebView?,
+              script: any AIChatJavascript.Type,
+              querySubmited: String? = nil) {
+    self.webView = webView
+    self.script = script
+    self.querySubmited = querySubmited
+    
+    super.init()
+
+    // Initialize
+    api = braveCore.aiChatAPI(with: self)
+    currentModel = api.currentModel
+    models = api.models
+    conversationHistory = api.conversationHistory
+    suggestedQuestions = api.suggestedQuestions
+    apiError = api.currentAPIError
+    
+    Task { @MainActor in
+      await getPremiumStatus()
+    }
+    
+    if isAgreementAccepted {
+      api.setConversationActive(true)
+    }
+  }
+  
+  public func getPageTitle() -> String? {
```

Comment:
doesn't need to be a func since it takes no args → `public var pageTitle: String? { ... }`